### PR TITLE
10012 add storage filename config

### PR DESCRIPTION
**Pull request by bferguso**
### Types of changes

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Description of Change

Adds the ability to configure the target filename and directory for uploaded files.

### Issues Solved

#10012

### Checklist

-   [x] Unit tests pass locally with my changes
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [ ] I have added necessary documentation (if appropriate)

#### Ticket Background
*   Sponsored by: Self Funded
*   Found by: @bferguso
*   Tested by: @bferguso 
*   Designed by: @bferguso

### CODE REVIEW

File: arches/app/templates/views/components/plugins/workflow.htm [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
         <div class="tabbed-workflow-title-bar">
             <div class="workflow-name-container">
-                <div class="workflow-name" data-bind="text: workflowName() + ' Workflow'">{% trans "Insert Workflow Name" %}</div>
```
Comment (aarongundel) [inline arches/app/templates/views/components/plugins/workflow.htm:10]: Is this change related to this PR?

File: arches/app/utils/storage_filename_generator.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import os
+from arches.app.models.system_settings import settings
+
+def generate_filename(instance, filename):
+    return os.sep.join([settings.UPLOADED_FILES_DIR, filename])
```
Comment (aarongundel) [inline arches/app/utils/storage_filename_generator.py:5]: Why use os.sep.join here?  os.path.join is a little smarter when doing these path joins.

**Review by aarongundel — CHANGES_REQUESTED** [review]
Looks good!  Best yet - works for me.  Some minor feedback here.  Can you also merge dev/7.5.x into your branch to clean up some of the noise?

File: arches/app/utils/storage_filename_generator.py (hunk 2 above)
Comment (bferguso) [reply arches/app/utils/storage_filename_generator.py]: @aarongundel - nice catch. Made the requested change and did a cherry pick on a clean dev/7.5.x to clean up the branch.

**Review by aarongundel — APPROVED** [review]
This looks great - thanks for the PR!
